src/rpc/mining.cpp

```diff
@@ -124,14 +123,17 @@ static UniValue generateBlocks(const CScript& coinbase_script, int nGenerate, ui
             LOCK(cs_main);
             IncrementExtraNonce(pblock, ::ChainActive().Tip(), nExtraNonce);
         }
-        while (nMaxTries > 0 && pblock->nNonce < nInnerLoopCount && !CheckProofOfWork(pblock->GetHash(), pblock->nBits, Params().GetConsensus())) {
+        while (nMaxTries > 0 && pblock->nNonce < std::numeric_limits<uint32_t>::max() && !CheckProofOfWork(pblock->GetHash(), pblock->nBits, Params().GetConsensus()) && !ShutdownRequested()) {
             ++pblock->nNonce;
             --nMaxTries;
         }
+        if (ShutdownRequested()) {
+            break;
+        }
```

Comment:
No biggie but you could just do
```C++
if (nMaxTries == 0 || ShutdownRequested()) {
    break;
}
```
i.e. merge the above and below blocks into one.

Reply:
nit picked